jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -393,103 +445,109 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
-            if isinstance(aperture, CircularAperture):
-                radii = fac * aperture.r  # radius
-            elif isinstance(aperture, EllipticalAperture):
-                radii = fac * aperture.a  # semimajor axis
-                radii_b = fac * aperture.b  # semiminor axis
-            elif isinstance(aperture, RectangularAperture):
-                radii = fac * aperture.w  # full width
-                radii_h = fac * aperture.h  # full height
-            else:
-                raise NotImplementedError(f"{aperture.__class__.__name__} is not supported")
-
-            mask_weights = np.zeros(flux_cube.shape, dtype=np.float32)
-
-            # Loop through cube and create cone aperture at each wavelength. Then convert that to a
-            # weight array using the selected aperture method, and add it to a weight cube.
-            for index, cone_r in enumerate(radii):
-                if isinstance(aperture, CircularAperture):
-                    aperture.r = cone_r
-                elif isinstance(aperture, EllipticalAperture):
-                    aperture.a = cone_r
-                    aperture.b = radii_b[index]
-                else:  # RectangularAperture
-                    aperture.w = cone_r
-                    aperture.h = radii_h[index]
-
-                slice_mask = aperture.to_mask(method=aper_method).to_image(im_shape)
-                # Add slice mask to fractional pixel array
-                mask_weights[:, :, index] = slice_mask
+    @with_spinner('bg_spec_spinner')
+    def extract_bg_spectrum(self, add_data=False, **kwargs):
+        """
+        Create a background 1D spectrum from the input parameters defined in the plugin.
+
+        If ``function`` is 'sum', then the value is scaled by the relative ratios of the area
+        (along the spectral axis) of ``aperture`` to ``background``.
+
+        Parameters
+        ----------
+        add_data : bool
+            Whether to add the resulting spectrum to the application, according to the options
+            defined in the plugin.
+        kwargs : dict
+            Additional keyword arguments passed to the NDDataArray collapse operation.
+            Examples include ``propagate_uncertainties`` and ``operation_ignores_mask``.
+        """
+        if self.background.selected != self.background.default_text:
+            bg_spec = self._extract_from_aperture(self.spectral_cube, self.uncert_cube,
+                                                  self.background, self.bg_wavelength_dependent,
+                                                  self.function_selected.lower(), **kwargs)
+            if self.function_selected.lower() == 'sum':
+                # then scale according to aperture areas across the spectral axis (allowing for
+                # independent wavelength-dependence btwn the aperture and background)
+                bg_spec *= self.aperture_area_along_spectral / self.bg_area_along_spectral
```

Comment:
@bmorris3 @camipacifici - normalization by aperture area is currently handled here.  Is this different from what you had in mind?